tests/Core/Ruleset/PopulateTokenListenersTest.php

```diff
@@ -0,0 +1,543 @@
+                'expected'     => 'first',
+            ],
+            'Sniff with multiple properties being set - first property'  => [
+                'sniffClass'   => 'PHP_CodeSniffer\\Standards\\Squiz\\Sniffs\\ControlStructures\\ForEachLoopDeclarationSniff',
+                'propertyName' => 'requiredSpacesAfterOpen',
+                'expected'     => '3',
+            ],
+            'Sniff with multiple properties being set - second property' => [
+                'sniffClass'   => 'PHP_CodeSniffer\\Standards\\Squiz\\Sniffs\\ControlStructures\\ForEachLoopDeclarationSniff',
+                'propertyName' => 'requiredSpacesBeforeClose',
+                'expected'     => '8',
+            ],
+        ];
+
+    }//end dataTriggersPropertySettingWhenPropertiesProvided()
+
+
+    /**
+     * Verifies that the "class" and "source" indexes get set.
+     *
+     * @return void
+     */
+    public function testSetsClassAndSourceIndexes()
+    {
+        foreach (self::$ruleset->tokenListeners as $token => $listeners) {
+            $this->assertTrue(is_array($listeners), 'No listeners registered for token'.Tokens::tokenName($token));
+
+            foreach ($listeners as $className => $details) {
+                $this->assertArrayHasKey(
+                    'class',
+                    $details,
+                    sprintf('"tokenizers" key missing for sniff class %s for token %s', $className, Tokens::tokenName($token))
+                );
+
+                $this->assertSame(
+                    $className,
+                    $details['class'],
+                    sprintf('Unexepcted value for "class" key for sniff class %s for token %s', $className, Tokens::tokenName($token))
+                );
+
+                $this->assertArrayHasKey(
+                    'source',
+                    $details,
+                    sprintf('"source" key missing for sniff class %s for token %s', $className, Tokens::tokenName($token))
+                );
+
+                $expected = '.'.substr($className, (strrpos($className, '\\') + 1), -5);
+
+                $this->assertStringEndsWith(
+                    $expected,
+                    $details['source'],
+                    sprintf('Unexepcted value for "source" key for sniff class %s for token %s', $className, Tokens::tokenName($token))
+                );
+            }//end foreach
+        }//end foreach
+
+    }//end testSetsClassAndSourceIndexes()
+
+
+    /**
+     * Verifies that sniffs by default are listening for PHP files only.
+     *
+     * @return void
+     */
+    public function testSetsSupportedTokenizersToPHPByDefault()
+    {
+        $exclude  = 'PHP_CodeSniffer\\Standards\\Generic\\Sniffs\\Files\\EndFileNewlineSniff';
+        $expected = ['PHP' => 'PHP'];
+
+        foreach (self::$ruleset->tokenListeners as $token => $listeners) {
+            $this->assertTrue(is_array($listeners), 'No listeners registered for token'.Tokens::tokenName($token));
```

Comment:
Was there a reason not to use `sprintf` here? Seems to be used in the other cases.

Reply:
There seem to be some other copies where this same construct is used, throughout this class.

Reply:
@NielsdeBlaauw Thanks for reviewing.

As for when interpolation vs concatenation vs `sprintf()`: there is no hard rule being followed. The important thing here is that if a test contains more than one assertion, the `$message` parameter is being set for each assertion to make it more straight forward to know which assertion failed (if the test would fail). Especially if there are, for instance, three `assertTrue()` assertions, it becomes hard to distinguish which failed - "false is not true" - without those `$message` parameters.

For all practical purposes, whether interpolation vs concatenation vs `sprintf()` is used for the `$message` parameter is largely a balance of code readability, trying to keep the line length below 120 and trying not to violate the "concatenation operator is not allowed to have whitespace around it" CS rule which is being enforced for this codebase.